### DISCUSSION TITLE
Error response from daemon

### DISCUSSION THREAD

**Discussion by superit63** [engageub/InternetIncome#147] [Q&A] [answered]
Hi,

I have faced the issue:
![image](https://github.com/user-attachments/assets/7c69cdd2-ba80-4559-9e97-f556da7322ef)
when I run command to start or delete the script. 
Thanks

**Comment by engageub** [comment]
Hello,
Could you please let me know which container were you trying to run. 
Is this main branch or test branch.

Thank you

**Reply by superit63** [reply]
Hi,
Its earnapp and I am using test branch.

**Reply by engageub** [reply]
Thank you for the confirmation. 

Can you try to restart all the earnapp containers using the following command and the delete them. 
Also, I think it does not occur while starting the script. instead while deleting it. 

`
sudo docker container ls -a | grep earnapp | awk '{print $1}'| xargs sudo docker restart
`


Thank you

**Reply by engageub** [reply]
Health Checks were disabled for earnapp in test branch as it is contributing to slight increase in cpu usage. 
Instead it is restarted every 24 hours. But the error message mentions to use --init option which is used for long running docker containers. This might be a permanent solution instead of temporarily restarting docker containers for health checks.

Thank you

**Reply by superit63** [reply]
Ahh, I got it. I will update the new code for Earnapp. I have used the old code since May. Thanks

**Reply by engageub** [reply]
Are you able to delete it after restarting the containers. If you see `dindearnapp` in `containernames.txt` file, you are using the latest version for earnapp.

Thank you

**Reply by engageub** [reply]
If restarting and deleting the containers does not work, then follow the  suggestion mentioned in docker forums to kill the zombie process.

https://forums.docker.com/t/cannot-do-a-kill-or-stop-or-anything-else-to-zombie-container/86201/2 

Thank you

**Reply by superit63** [reply]
I got this error when I excute delete container:
![image](https://github.com/user-attachments/assets/33e32527-40f7-4413-a54b-b44580a1f317)
Is it possible to run docker stop $(docker ps -aq) and docker rm $(docker ps -aq)? I have tried and it works
Thanks

**Reply by engageub** [reply, marked as answer]
Hello,
There are two solutions here based on the above discussion. 

1. Start the containers using `--init` option as suggested by the docker error message. 
2. Stop and Delete the containers as it was done in earlier versions. The wiki still mentions in main page that we stop and delete containers. This was taking longer time to stop and delete the containers, so `docker rm -f` command is being used to force delete the containers immediately. 

However, it is still not clear why this is happening on your PC and not mine.
Could you please let me know which version of docker you are using and also the architecture whether it is arm64 or amd64.
I have been running earnapp for about a month and did not have problem with `--delete` command. 

Could you please search for `docker run` in the code and replace all of them with `docker run --init` and then start the containers.
Please confirm if you are able to delete them after starting. 


Thank you

**Reply by superit63** [reply]
Thanks a lot for your help. I can fix it with --init. I get these error after starting.
Another option is upgrading the hardware of machine as 1 app can consume around 100-150mb RAM

**Reply by superit63** [reply]
I have seen the logs and see that:
2024-08-19 14:24:03.478 NOTICE: perr https://perr.l-err.biz/perr?id=lum_sdk_node_heartbeat_proxy_js_conn_02_4min&appid=node_earnapp.com&ver=1.294.218&ver_sdk=1.294.218&uuid=sdk-node-cd333f8649d0e5d6513351fb57b077bc res 200
2024-08-19 14:24:03.513 NOTICE: perr https://perr.l-err.biz/perr?id=lum_sdk_node_heartbeat_proxy_js_conn_02_4min&appid=node_earnapp.com&ver=1.294.218&ver_sdk=1.294.218&uuid=sdk-node-cd333f8649d0e5d6513351fb57b077bc res 200
2024-08-19 14:25:03.009 NOTICE: idle_state_monitor run
2024-08-19 14:26:04.049 NOTICE: idle_state_monitor run
2024-08-19 14:27:05.089 NOTICE: idle_state_monitor run
2024-08-19 14:28:03.479 ERR: perr heartbeat_proxy_js_conn_03_8min {}
2024-08-19 14:28:03.480 NOTICE: perr lum_sdk_node_heartbeat_proxy_js_conn_03_8min proxy
2024-08-19 14:28:03.517 ERR: perr heartbeat_proxy_js_conn_03_8min {}
2024-08-19 14:28:03.518 NOTICE: perr lum_sdk_node_heartbeat_proxy_js_conn_03_8min proxy
2024-08-19 14:28:04.126 NOTICE: perr https://perr.l-err.biz/perr?id=lum_sdk_node_heartbeat_proxy_js_conn_03_8min&appid=node_earnapp.com&ver=1.294.218&ver_sdk=1.294.218&uuid=sdk-node-cd333f8649d0e5d6513351fb57b077bc res 200
2024-08-19 14:28:04.264 NOTICE: perr https://perr.l-err.biz/perr?id=lum_sdk_node_heartbeat_proxy_js_conn_03_8min&appid=node_earnapp.com&ver=1.294.218&ver_sdk=1.294.218&uuid=sdk-node-cd333f8649d0e5d6513351fb57b077bc res 200
2024-08-19 14:28:06.129 NOTICE: idle_state_monitor run
2024-08-19 14:29:07.169 NOTICE: idle_state_monitor run
2024-08-19 14:30:01.705 NOTICE: status_get: {"idle":true,"usage":{},"idle_state":{"mem_usage":5,"mem_usage_ts":1724077747190,"cpu_usage":0.34,"cpu_usage_ts":1724077748208},"bw":{}}
2024-08-19 14:30:01.716 NOTICE: status_get: {"idle":true,"usage":{},"idle_state":{"mem_usage":5,"mem_usage_ts":1724077747190,"cpu_usage":0.34,"cpu_usage_ts":1724077748208},"bw":{}}
2024-08-19 14:30:08.209 NOTICE: idle_state_monitor run
2024-08-19 14:31:09.249 NOTICE: idle_state_monitor run
It seems that my machine specs is not good enough. I suggest to run 100 earnapp in machine that has at least 32gb RAM to have the best performance and good earning. How do you think?

**Reply by engageub** [reply]
Thank you for response. It is not clear whether adding `--init` actually resolved the issue.
In addition to this, could you please elaborate what you mean by "upgrading the hardware of machine as 1 app" .
Do you mean adding all the apps into single docker image?

Thank you

**Reply by engageub** [reply]
> I have seen the logs and see that: 2024-08-19 14:24:03.478 NOTICE: perr https://perr.l-err.biz/perr?id=lum_sdk_node_heartbeat_proxy_js_conn_02_4min&appid=node_earnapp.com&ver=1.294.218&ver_sdk=1.294.218&uuid=sdk-node-cd333f8649d0e5d6513351fb57b077bc res 200 2024-08-19 14:24:03.513 NOTICE: perr https://perr.l-err.biz/perr?id=lum_sdk_node_heartbeat_proxy_js_conn_02_4min&appid=node_earnapp.com&ver=1.294.218&ver_sdk=1.294.218&uuid=sdk-node-cd333f8649d0e5d6513351fb57b077bc res 200 2024-08-19 14:25:03.009 NOTICE: idle_state_monitor run 2024-08-19 14:26:04.049 NOTICE: idle_state_monitor run 2024-08-19 14:27:05.089 NOTICE: idle_state_monitor run 2024-08-19 14:28:03.479 ERR: perr heartbeat_proxy_js_conn_03_8min {} 2024-08-19 14:28:03.480 NOTICE: perr lum_sdk_node_heartbeat_proxy_js_conn_03_8min proxy 2024-08-19 14:28:03.517 ERR: perr heartbeat_proxy_js_conn_03_8min {} 2024-08-19 14:28:03.518 NOTICE: perr lum_sdk_node_heartbeat_proxy_js_conn_03_8min proxy 2024-08-19 14:28:04.126 NOTICE: perr https://perr.l-err.biz/perr?id=lum_sdk_node_heartbeat_proxy_js_conn_03_8min&appid=node_earnapp.com&ver=1.294.218&ver_sdk=1.294.218&uuid=sdk-node-cd333f8649d0e5d6513351fb57b077bc res 200 2024-08-19 14:28:04.264 NOTICE: perr https://perr.l-err.biz/perr?id=lum_sdk_node_heartbeat_proxy_js_conn_03_8min&appid=node_earnapp.com&ver=1.294.218&ver_sdk=1.294.218&uuid=sdk-node-cd333f8649d0e5d6513351fb57b077bc res 200 2024-08-19 14:28:06.129 NOTICE: idle_state_monitor run 2024-08-19 14:29:07.169 NOTICE: idle_state_monitor run 2024-08-19 14:30:01.705 NOTICE: status_get: {"idle":true,"usage":{},"idle_state":{"mem_usage":5,"mem_usage_ts":1724077747190,"cpu_usage":0.34,"cpu_usage_ts":1724077748208},"bw":{}} 2024-08-19 14:30:01.716 NOTICE: status_get: {"idle":true,"usage":{},"idle_state":{"mem_usage":5,"mem_usage_ts":1724077747190,"cpu_usage":0.34,"cpu_usage_ts":1724077748208},"bw":{}} 2024-08-19 14:30:08.209 NOTICE: idle_state_monitor run 2024-08-19 14:31:09.249 NOTICE: idle_state_monitor run It seems that my machine specs is not good enough. I suggest to run 100 earnapp in machine that has at least 32gb RAM to have the best performance and good earning. How do you think?

The logs show cpu usage as 0.34. Do you not see any earnings.
Also, have you checked the cpu usage using `top` command.  Based on that you can decide the number of containers that can be run on a single machine. Make sure your network has enough bandwidth. 


Thank you

**Reply by superit63** [reply]
Thanks,
I mean I got these error when I run 100 earnapp (100 docker images) in machine 8gb RAM, then I upgrade my machine to 64gb RAM (ofc, I run 200 proxies/200 Earnapp in this machine), that errors dont occurs; however, the CPU usage is low and I see very little earning.
Until now, I dont know why I cant get much earning (compare to run only 10 proxy/machine 8gb RAM). I just supposed that the specs of machine should be strong to run more apps.
The last thing is my network has enough bandwidth because I am using dedicated server with 1 gpbs dedicated port

**Reply by engageub** [reply]
> Thanks, I mean I got these error when I run 100 earnapp (100 docker images) in machine 8gb RAM, then I upgrade my machine to 64gb RAM (ofc, I run 200 proxies/200 Earnapp in this machine), that errors dont occurs; however, the CPU usage is low and I see very little earning. Until now, I dont know why I cant get much earning (compare to run only 10 proxy/machine 8gb RAM). I just supposed that the specs of machine should be strong to run more apps. The last thing is my network has enough bandwidth because I am using dedicated server with 1 gpbs dedicated port

Thank you for the detailed response. So you have not used  `--init` command on the old host. 
Are you using `--init` on the new host with 64gb RAM. 

The earnings may depend on demand in the location. Also yesterday was weekend. 

I have concern about the following error. Does it happen normally with direct connection as well.

```
2024-08-19 14:28:03.517 ERR: perr heartbeat_proxy_js_conn_03_8min {}
```

**Reply by superit63** [reply]
Hi, I am not using --init on the new host with 64gb RAM. Everything is good.
Related to earning, I know that it depends on the demand in location; but I dont know why it decreased a lot and I see many errors like you concerned:
`2024-08-19 14:28:03.517 ERR: perr heartbeat_proxy_js_conn_03_8min {}`

**Reply by engageub** [reply]
Thank you for the response. 
I had a look at my PC with direct connection and these errors were also present last year. 
So, they seem to be common.
If you see your devices online and green, it indicates they have already done checks. 

Thank you

**Reply by engageub** [reply]
You can you try changing the default DNS to 1.1.1.1 instead of 8.8.8.8 to see if it makes any difference. 
You will have to replace line 233 with the following.

```
printf 'nameserver 1.1.1.1\nnameserver 1.0.0.1\nnameserver 9.9.9.9\nnameserver 8.8.8.8\nnameserver 8.8.4.4\n' > $dns_resolver_file;

```

Thank you

**Comment by engageub** [comment]
Hello,
I would like to let you know that new options `--stop` and `--restart` have been added to the test branch.
In case you have issues with zombie process, please use `--stop` and then `--delete` .
Using `--init` option is creating a new process which consumes slight CPU utilization. 
Ideally zombie process should not be created while using docker container. 

Thank you

**Comment by engageub** [comment]
An additional option `USE_DOCKER_INIT` has been added to properties file to start the containers with `--init` option if required. It is disabled by default.
Both the solutions mentioned above have been added to the code in test branch.
